src/rules/store/avoid-mapping-selectors.ts

```diff
@@ -29,16 +34,15 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
     const storeName = findNgRxStoreName(context)
     if (!storeName) return {}
 
-    const mapInSelect = `CallExpression:has(MemberExpression:has(${storeSelect(
+    const pipeWithSelectAndMapSelector = `${storePipe(
       storeName,
-    )}))[callee.property.name='pipe']`
-    const mapInPipe = `${storePipe(
+    )}:has(CallExpression[callee.name='select'] ~ CallExpression[callee.name='map'])`
+    const selectSelector = `${storeExpressionCallable(
       storeName,
-    )}:has(CallExpression[callee.name="select"])`
-    const pipeExpressions = [mapInSelect, mapInPipe].join(', ')
+    )}[callee.object.callee.property.name='select']`
 
     return {
-      [`:matches(${pipeExpressions}) > CallExpression[callee.name='map']`](
+      [`:matches(${selectSelector}, ${pipeWithSelectAndMapSelector}) CallExpression[callee.name='map']`](
```

Comment:
I think it worths to mention that the following case is not covered by this selector:

<img width="1440" alt="subsequent pipes" src="https://user-images.githubusercontent.com/11965907/131405406-13464f15-e1cb-4e2f-9bb2-80d6c308564c.png">

But I don't think people really use it like this in real world 🤷🏼‍♂️.

Reply:
Yea... we're not going to support that 😅

Reply:
```suggestion
      [`:matches(${selectSelector}, ${pipeWithSelectAndMapSelector}) > CallExpression[callee.name='map']`](
```